java/memory/src/main/java/org/apache/arrow/memory/Accountant.java

```diff
@@ -78,27 +78,36 @@ public Accountant(Accountant parent, long reservation, long maxAllocation) {
         throw new OutOfMemoryException(String.format(
             "Failure trying to allocate initial reservation for Allocator. " +
                 "Attempted to allocate %d bytes and received an outcome of %s.", reservation,
-            outcome.name()));
+            outcome.getStatus().name()), outcome.getDetails());
       }
     }
   }
 
   /**
    * Attempt to allocate the requested amount of memory. Either completely succeeds or completely
-   * fails. Constructs a a
-   * log of delta
-   *
-   * <p>If it fails, no changes are made to accounting.
+   * fails. If it fails, no changes are made to accounting.
    *
    * @param size The amount of memory to reserve in bytes.
-   * @return True if the allocation was successful, false if the allocation failed.
+   * @return the status and details of allocation at each allocator in the chain.
    */
   AllocationOutcome allocateBytes(long size) {
-    final AllocationOutcome outcome = allocate(size, true, false);
-    if (!outcome.isOk()) {
+    AllocationOutcomeDetails details = null;
```

Comment:
it seems like it might be cleaner to overload.  Is there a reason not to so?

Reply:
added an overloaded fn - but i'm not sure that's what you meant. please take a look.